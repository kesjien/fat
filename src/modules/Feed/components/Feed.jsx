import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Panel, Row, Input } from 'muicss/react'
// important here because of the bootstraping and speed, muicss input has bigger priority than
// rule: avoid important if possible
// inline styled
// in real world use another or custom implementation of inputs
const StyledRow = styled(Row)`
  margin: 1em !important; 
`

const StyledPanel = styled(Panel)`
  margin: 1em;
`
export class Feed extends React.Component {
  // define data model for better validation
  static propTypes = {
    data: PropTypes.object, // eslint-disable-line
    changeField: PropTypes.func.isRequired
  }
  static defaultProps = {
    data: {}
  }
  constructor (props) {
    super(props)
    // example of the statuses handling, should be moved to constants and
    // implemented as a list of the status
    this.cancelStatus = 'PRE_AUTHORIZED_CANCELLED'
  }

  render () {
    const { changeField, data } = this.props

    const feedComponent = []
    if (data) {
      Object.keys(data).forEach((key) => {
        if (key === 'fromDate' || key === 'toDate') {
          data[key] = data[key].slice(0, 10)
        } else if (key === 'status') { // handle status here
          // handle and parse data in the separate function, move to separate function
          if (data[key] === this.cancelStatus) {
            data[key] = 'Cancelled'
          }
        } else if (data[key] === null) {
          data[key] = ''
        }
        feedComponent.push(
          <StyledRow key={key}>
            <Input label={key} name={key} value={data[key]} onChange={changeField} />
          </StyledRow>
        )
      })
    }
    return (
      <StyledPanel>
        {feedComponent}
      </StyledPanel>
    )
  }
}

