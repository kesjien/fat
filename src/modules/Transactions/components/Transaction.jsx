import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Col, Row } from 'muicss/react'

const StyledRow = styled(Row)`
  border: 1px dotted #cccccc5e;
  min-height: 40px;
  text-align: center;
  margin-left: 0px !important;
  margin-right: 0px !important;
  cursor: pointer;
  &:hover {
    background-color: #f0f5fa;
  }
`

const StyledDiv = styled.div`
  border-right: 1px solid #cccccc42;
`

const StyledCol = styled(Col)`
  padding-left: 0px !important;
  padding-right: 0px !important;
`

const Transaction = (props) => {
  const cols = []
  Object.keys(props.data).forEach((key, i) => {
    cols.push(<StyledCol key={`${i}_key`} md="2"><StyledDiv className="mui--appbar-height mui--appbar-line-height">{props.data[key]}</StyledDiv></StyledCol>)
  })

  return (
    <StyledRow>
      {cols}
    </StyledRow>
  )
}

Transaction.propTypes = {
  data: PropTypes.oneOfType([
    PropTypes.array, // eslint-disable-line
    PropTypes.object // eslint-disable-line
  ]).isRequired
}

export default Transaction
