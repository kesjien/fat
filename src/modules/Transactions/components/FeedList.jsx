import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { Panel } from 'muicss/react'
import PropTypes from 'prop-types'
import Transaction from './Transaction'

const PanelComponent = styled.div`
  margin: 1em;
  padding: 3em;
`
const StyledLink = styled(Link)`
  color: #000;
`
const Header = styled.div`
  background-color: #e6f4ff;
`

export class FeedList extends React.PureComponent {
  static propTypes = {
    // define model schema for validation transactions. temporary eslint omit line
    transactions: PropTypes.array.isRequired, // eslint-disable-line
  }
  render () {
    const transactions = []
    let newData = {}
    this.props.transactions.forEach((item) => {
      newData = {
        borrowerId: item.borrowerId,
        creditUsed: item.creditUsed,
        currency: item.currency,
        fromDate: item.fromDate.slice(0, 10),
        lenderId: item.lenderId,
        price: item.price
      }
      transactions.push(
        <div key={item.id}>
          <StyledLink to={`/${item.id}`}>
            <Transaction data={newData} />
          </StyledLink>
        </div>
      )
    })
    return (
      <Panel>
        <PanelComponent>
          <Header>
            <Transaction data={Object.keys(newData)} />
          </Header>
          {transactions}
        </PanelComponent>
      </Panel>
    )
  }
}
