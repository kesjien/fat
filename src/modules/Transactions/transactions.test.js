import React from 'react'
import renderer from 'react-test-renderer'
import Transaction from './components/Transaction'
import { FeedList } from './components/FeedList'

it('renders correctly', () => {
  const tree = renderer
    .create(<Transaction data={[]} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly', () => {
  const tree = renderer
    .create(<FeedList transactions={[]} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
