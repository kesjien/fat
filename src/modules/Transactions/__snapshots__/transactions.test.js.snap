// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mui-row sc-bdVaJa bPYCzj"
/>
`;

exports[`renders correctly 2`] = `
<div
  className="mui-panel "
>
  <div
    className="sc-bxivhb iuYwsc"
  >
    <div
      className="sc-EHOje jRFSxW"
    >
      <div
        className="mui-row sc-bdVaJa bPYCzj"
      />
    </div>
  </div>
</div>
`;
